File: Intersection of two Integer arrays/Intersection.py
#Written by Kenneth Nero <https://github.com/KennethNero>

# Simple function to create an array of mathcing elements
# between two lists.
def intersection(ls1, ls2):
    result = [v for v in ls1 if v in ls2]
    return result

# Main function that operates on the command line to take
# in and send output. 
def main():
    ls = []
    inp = input("Please enter your first comma seperated list of integers: ")
    inp2 = input("Please enter your second comma seperated list of integers: ")

    result = intersection(inp.replace(' ', '').split(','),
                          inp2.replace(' ','').split(','))
    print("\nIntersection between first and second list:")
    outStr = "["
    for x in result:
        outStr = outStr+str(x)+', '
    print(outStr[0:-2]+']')

if __name__ == "__main__":
    main()

File: Intersection of two Integer arrays/Intersection.cpp
#include<iostream>
using namespace std;
int main(){
	int n1=0;
	int n2=0;
	int a1[100];
	int a2[100];
	int a3[100];
	int i=0;
	int j=0;
	int k=0;
	int cnt=0;
	cout<<"Enter the size for 1st array: ";
	cin>>n1;
	cout<<"Enter the size for 2nd array: ";
	cin>>n2;
	cout<<"Enter the elements for 1st array: ";
	for(i=0;i<n1;i++){
		cin>>a1[i];
	}
	cout<<"Enter the elements for 2nd array: ";
	for(i=0;i<n2;i++){
		cin>>a2[i];
	}
	for(i=0;i<n1;i++){
		for(j=0;j<n2;j++){
		if(a1[i]==a2[j]){
			a3[k]=a1[i];
			k++;
			cnt++;
		}}}
		if(cnt==0){
			cout<<"There are no common elements the two Arrays\n";
		}
		cout<<"The intersection Array of the above arrays is: ";
	for(k=0;k<cnt;k++){
		cout<<a3[k]<<"\t";
	}
}
